astropy/wcs/wcs.py

```diff
@@ -523,6 +523,8 @@ def __init__(
     ):
         close_fds = []
 
+        self._preserve_units = False if preserve_units is None else preserve_units
```

Comment:
What does `preserve_units=None` do? It is not documented anywhere. Why a user would pass `preserve_units=None` to the initializer?

Also, in the docstring:

```
    preserve_units : str, optional
        By default, some units are converted to SI, for example spectral
        axes in units of nm might be converted to m, and celestial axes
        in units of arcsec might be converted to deg. If ``preserve_units``
        is set to `True`, the original units will be preserved.
```
What type is the `preserve_units`? is it `str` or `bool`? and `None`? That is, should it be:
```
    preserve_units : bool, None, optional
```
and add a description for `None`?